bedrock/cms/decorators.py

```diff
@@ -33,42 +68,111 @@ def prefer_cms(view_func):
         def some_path(request):
             ...
 
     Or, in a URLconf for a regular Django view:
 
         ...
-        path("some/path/", prefer_cms(views.some_view)),
+        path("some/path/", prefer_cms(views.some_view, fallback_ftl_files=[...])),
+
+        # or
+
+        path("some/path/", prefer_cms(views.some_view, fallback_lang_codes=["fr", "pt-BR",])),
+
         ...
 
-    Or, in a URLconf with Bedrock's page() helper:
-        page(
-            "about/leadership/",
-            "mozorg/about/leadership/index.html",
-            decorators=[prefer_cms],
-        )
+        # or
+
+        path("some/path/", prefer_cms(views.some_view, fallback_callable=path.to.callable)),
+
+        ...
+
+    IMPORTANT: there's no support for bedrock.mozorg.util.page(), deliberately.
+
+        Making prefer_cms work with our page() helper would have made that function
+        more complex. Given that it's straightforward to convert a page()-based view to
+        a dedicated TemplateView, which _can_ be decorated with prefer_cms at the
+        URLConf level, that is the recommended approach if you are migrating a page()
+        based view to the CMS, or need to have hybrid behaviour.
 
     """
 
-    @wraps(view_func)
-    def wrapped_view(request, *args, **kwargs):
-        path = remove_lang_prefix(request.path_info)
-        try:
-            # Does Wagtail have a route that matches this? If so, show that page
-            wagtail_response = wagtail_serve(request, path)
-            if wagtail_response.status_code == HTTP_200_OK:
-                return wagtail_response
-        except Http404:
-            pass
-
-        # If not, call the original view function, but remember to
-        # un-mark the request as being for a CMS page. This is so to avoid
-        # lib.l10n_utils.render() incorrectly looking for available translations
-        # based on CMS data - we're falling back to non-CMS pages, so we want
-        # the Fluent translations that are normal for a Django-rendered page
-        request.is_cms_page = False
-        return view_func(request, *args, **kwargs)
-
-    return wrapped_view
+    if len([x for x in [fallback_ftl_files, fallback_lang_codes, fallback_callable] if x]) > 1:
+        raise RuntimeError(
+            "The prefer_cms decorator can be configured with only one of fallback_ftl_files or fallback_lang_codes or fallback_callable."
+        )
+
+    fallback_ftl_files = fallback_ftl_files or []
+    fallback_lang_codes = fallback_lang_codes or []
+
+    def _get_django_locales_available(
+        *,
+        fallback_ftl_files,
+        fallback_lang_codes,
+        fallback_callable,
+        kwargs,
+    ):
+        # Prefer explicit callable to get lang codes
+        if fallback_callable:
+            return fallback_callable(**kwargs)
+
+        # Use explicit list of lang codes over fluent files
+        if fallback_lang_codes:
+            return fallback_lang_codes
```

Comment:
**note (non-blocking):** The `fallback_callable` approach was a late addition in order to make the VPN RC (with its inconsistent Django-side db-backed localization) work -- it's quite an outlier of a view for Bedrock. 

It may be that the `fallback_callable` and the `fallback_ftl_files` are actually the only two approaches we need and we ca drop `fallback_lang_codes`, because they can be generated from a callable - let's see as we use `prefer_cms` more